components/server/src/authorization/spicedb-authorizer.ts

```diff
@@ -14,6 +14,30 @@ import { SpiceDBClientProvider } from "./spicedb";
 import * as grpc from "@grpc/grpc-js";
 import { isFgaChecksEnabled } from "./authorizer";
 
+async function tryThree<T>(errMessage: string, code: (attempt: number) => Promise<T>): Promise<T> {
+    let attempt = 0;
+    // we do sometimes see INTERNAL errors from SpiceDB, so we retry a few times
+    // last time we checked it was 15 times per day (check logs)
+    while (attempt++ < 3) {
+        try {
+            return await code(attempt);
+        } catch (err) {
+            if (err.code === grpc.status.INTERNAL && attempt < 3) {
```

Comment:
just asking did we report internal errors to spicedb? it could be also bugs which should be addressed by them?

Reply:
Of course, this is not a solution. But I want to shield the user experience from this until I've found the underlying issue.